lib/Driver/ToolChains.cpp

```diff
@@ -1597,7 +1597,7 @@ toolchains::GenericUnix::constructInvocation(const LinkJobAction &job,
     Arguments.push_back(context.Args.MakeArgString(A->getValue()));
   }
 
-  if (getTriple().getOS() == llvm::Triple::Linux) {
+  if (getTriple().getOS() == llvm::Triple::Linux && job.getKind() != LinkKind::DynamicLibrary) {
```

Comment:
I think this should be a positive check for Executable rather than a negative check for DynamicLibrary. We could easily add StaticLibrary at some point.

(Also, nitpick: please stay within 80 columns. We'd usually wrap after the `&&` and indent to the open paren.)

Reply:
@jrose-apple fixed both. Good suggestion on the positive vs negative test.